### PR TITLE
feat: Auto-Discovery should work for unauthorized devices

### LINKED ISSUE #3: Discovery fails for unauthorized devices
**Observation**
When discovering devices, if the device is password-protected, and the credentials do not match the default credentials, the device will not be added to EdgeX. This is because the `GetDeviceInformation` requires authentication.

**Problem**
Without these devices being added to EdgeX there is no way to assign credentials to them. Thus, currently your solution requires all auto-discovered cameras to share the same username, password, and authentication type (which is bad practice).

**Proposed Solution**
- If a device fails the `GetDeviceInformation`, but it is still a valid ONVIF device, it should be added to EdgeX, so that the credentials can be added after the fact.
- Fix #2 and you will be able to add devices with unique names even without authentication.

### PR DESCRIPTION
**Pull request by weichou1229**
Close #3

Signed-off-by: bruce <weichou1229@gmail.com>

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features) **not impact the test**
- [ ] Docs have been added / updated (for bug fixes / features) **not impact the doc**

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-examples/blob/master/.github/CONTRIBUTING.md.

## What is the current behavior?

If a device fails the GetDeviceInformation, the device can't be added to the EdgeX.

## What is the new behavior?
If a device fails GetDeviceInformation, it should be added to EdgeX, so that the credentials can be added after the fact.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports
Are there any new imports or modules? If so, what are they used for and why?

## Specific Instructions
Are there any specific instructions or things that should be known prior to reviewing?

## Other information

### CODE REVIEW

File: internal/driver/driver.go [hunk 1]
```diff
@@ -318,23 +318,31 @@ func (d *Driver) Discover() {
...
-			d.lc.Errorf("failed to get the device information for the camera %s, %v", dev.Name, edgexErr)
-			return
+			d.lc.Warnf("failed to get the device information for the camera %s, %v", endpointRef, edgexErr)
+			dev.Protocols[OnvifProtocol][SecretPath] = endpointRef
+			discovered = sdkModel.DiscoveredDevice{
+				Name:        endpointRef,
+				Protocols:   dev.Protocols,
+				Description: "Auto discovered Onvif camera",
+				Labels:      []string{"auto-discovery"},
+			}
```
Comment (ajcasagrande) [inline internal/driver/driver.go:331]:
```suggestion
			}
			d.lc.Debugf("Discovered unknown camera from the address '%s'", onvifDevice.GetDeviceParams().Xaddr)
```

**Review by ajcasagrande — COMMENTED** [review]
Looks good. One suggestion.

File: internal/driver/driver.go (hunk 1 above)
Comment (weichou1229) [reply internal/driver/driver.go:331]: Added the debug log.

**Review by ajcasagrande — APPROVED** [review]
LGTM

**Pull request merged**